lib/vagrant-service-manager/services/open_shift.rb

```diff
@@ -60,6 +63,30 @@ def self.print_info(ui, options)
         end
       end
 
+      def self.install_cli(machine, ui, options)
+        Installer.new('openshift', machine, ui, options).run
+      end
+
+      def self.binary_path(version)
+        "#{BIN_FOLDER}/openshift/#{version}/oc"
+      end
+
+      def self.binary_download_url(machine, version)
+        arch = machine.env.host.capability(:os_arch)
+        sha = version_sha(version)
+        download_base_path = "#{BINARY_BASE_URL}/download/v#{version}/"
+        file = "#{FILE_PREFIX}-v#{version}-#{sha}-#{os_type}-#{ext(machine, os_type)}"
+        download_base_path + file
+      end
+
+      def self.download_binary(url, path)
+        Dir.mkdir(TEMP_DIR) unless File.directory? TEMP_DIR
+        Vagrant::Util::Downloader.new(url, "#{TEMP_DIR}/#{File.basename(url)}").download!
+        dir = File.basename(url).sub('.tar.gz', '')
+        system("tar -xf #{TEMP_DIR}/#{File.basename(url)} -C #{TEMP_DIR}")
```

Comment:
this might not work on Windows. It is not a good idea to delegate to system command. Ruby has built-in classes for this - http://stackoverflow.com/questions/856891/unzip-zip-tar-tag-gz-files-with-ruby